dockers/base-cuda/Dockerfile

```diff
@@ -13,13 +13,14 @@
 # limitations under the License.
 
 ARG UBUNTU_VERSION=20.04
-ARG CUDA_VERSION=11.6.1
+ARG CUDA_VERSION=11.7.1
```

Comment:
This PR should also update the version used in gpu ci. Otherwise we arent sure if the image is working well

Reply:
well, some integration like collesalai has not been built yet for this CUDA version :otter:

Reply:
But we already have two separate GPU ci jobs. You just need to update the latest one

Reply:
This: https://github.com/Lightning-AI/lightning/blob/acdb1453f1a85441083f9b20e9d150cd6e9e8fd9/.azure/gpu-tests-pytorch.yml#L50

Also for Lite

Reply:
well, it seems that colossalAI is not built for PT 1.13 anyway
https://www.colossalai.org/download/

Reply:
But `colossalai` is not installed in the "PyTorch - latest" job anyways. Only the `PyTorch & strategies`

Reply:
true :)